tools/c7n_azure/c7n_azure/filters.py

```diff
@@ -478,3 +480,78 @@ def _check_rules(self, resource_rules):
             return self.policy_include.issubset(resource_rules)
         else:  # validated earlier, can never happen
             raise FilterValidationError("Internal error.")
+
+
+class ResourceLockFilter(Filter):
+    """Filter locked resources.
+     Lock can be of 2 types: ReadOnly and CanNotDelete. To filter any lock, use "Any".
+     Lock type is optional, by default any lock will be applied to the filter.
+
+     Example of policy that gets keyvaults with ReadOnly lock:
+
+    .. code-block :: yaml
+
+       policies:
+        - name: lock-filter
+          resource: azure.keyvault
+          filters:
+            - type: resource-lock
+              lock-type: ReadOnly
+
+    Example of policy that gets locked sqldatabases (any type of lock):
+
+    .. code-block :: yaml
+
+       policies:
+        - name: lock-filter
+          resource: azure.sqldatabase
+          filters:
+            - type: resource-lock
+
+    Example of policy that gets all unlocked resource groups:
+
+    .. code-block :: yaml
+
+       policies:
+        - name: lock-filter
+          resource: azure.resourcegroup
+          filters:
+            - not:
+                - type: resource-lock
+
+    """
+
+    schema = type_schema(
+        'resource-lock', required=['type'],
+        **{
+            'lock-type': {'type': 'string'}
```

Comment:
please use an enum to identify the possible lock forms, also it would be nice if this had direct support for finding resources without locks instead of having to use a not enclosure via perhaps lock-type: 'absent'